src/index.ts

```diff
@@ -1,127 +1,268 @@
+	if(fade_in == 0 && fade_out == 0) {
+		return '';
+	} else {
+		return `\\fad(${fade_in},${fade_out})`;
+	}
 }
 
-function sortStartTime(a: any, b: any) {
-	if (a.value.Start < b.value.Start) return -1;
-	if (a.value.Start > b.value.Start) return 1;
-	return 0;
+function displayremoveToDisplayRemove(displayremove: string) {
+	let [display, remove] = displayremove.split(',').map(x=>parseInt(x));
+	if(remove === undefined) remove=display;
+	return [display, remove]
 }
 
-function getStyleAss(style: StyleElement) {
-	return {
-		key: 'Style',
-		value: {
-			'Name': 'Default',
-			'Fontname': 'Arial',
-			'Fontsize': '24',
-			'PrimaryColour': '&H00FFFFFF',
-			'SecondaryColour': '&H000088EF',
-			'OutlineColour': '&H00000000',
-			'BackColour': '&H00666666',
-			'Bold': '-1',
-			'Italic': '0',
-			'Underline': '0',
-			'StrikeOut': '0',
-			'ScaleX': '100',
-			'ScaleY': '100',
-			'Spacing': '0',
-			'Angle': '0',
-			'BorderStyle': '1',
-			'Outline': '1.5',
-			'Shadow': '0',
-			'Alignment': '8',
-			'MarginL': '0',
-			'MarginR': '0',
-			'MarginV': '20',
-			'Encoding': '1',
-			...style
-		}
-	};
-}
+async function mainCLI() {
 
-/** Convert KBP data (txt) to ASS */
-export function convertToASS(time: string, options: SyllabesConfig): string {
-	const kbp = new KBPParser(options);
-	const kara = kbp.parse(time);
-	const dialogues = [];
-	const comments = [];
-
-	const styles = clone(ass.styles);
-	styles.body = styles.body.concat(kbp.styles.length > 0 ? kbp.styles.map(style => getStyleAss(style)) :  [ass.defaultStyle]);
-	const script = clone(ass.dialogue);
-	script.value.Effect = ass.scriptFX;
-	script.value.Text = ass.script;
-	script.key = 'Comment';
-	comments.push(script);
-	for (const line of kara.track) {
-		const ASSLines = generateASSLine(line, options);
-		comments.push(clone(ASSLines.comment));
-		dialogues.push(clone(ASSLines.dialogue));
+  // TODO: Either 1) only read the file in if track_offset is needed or 2) find other necessary settings to read
+  let track_offset = 0;
+  if ('APPDATA' in process.env ) {
+	try {
+		const settings = readFileSync(process.env.APPDATA + '/Karaoke Builder/data_studio.ini', 'utf8');
+		track_offset = parseFloat(settings.match(/^setoffset\s+(\S+)/m)[1]) / 100;
+	} catch (error) {
+		
 	}
-	comments.sort(sortStartTime);
-	dialogues.sort(sortStartTime);
-	const events = clone(ass.events);
-	events.body = events.body.concat(comments, dialogues);
-	return stringify([ass.scriptInfo, styles, events]);
-}
 
-async function mainCLI() {
-	if (!process.argv[2]) {
-		throw `KBP2ass - Convert KBP karaoke to ASS file
-		Usage: kbp2ass myfile.txt
-		Output goes to stdout
-		`;
-	}
-	const txtFile = process.argv[2];
+  }
+
+	// Per yargs docs, use of terminalWidth() in typescript requires workaround
+	// of assigning a variable name to the instance so it can be referenced
+	const yargsInstance = yargs(hideBin(process.argv))
+
+	const argv = yargsInstance
+		.scriptName('kbp2ass')
```

Comment:
Quotes are not very consistent, do you want to use double or single ?
Same for indent do you want to have 2 or 4 space or use tabs ?

Reply:
Good point. I think your original code was tabs, so I can standardize on that throughout. I think it's mostly there except for the types file which ended up with a strange 2 space indentation, with tabs instead of 4 spaces. There were probably a few places where my normal editor config of 4-space indents leaked in that I can fix as well.

As far as quotes, I can fix that too. Do you have a preference?

Reply:
I went ahead with double

Reply:
no problem for me but can you update to "double" the eslint config https://github.com/ItMightBeKaraoke/kbp2ass/blob/main/.eslintrc.json#L26 ?

Reply:
This should be properly fixed now:

`$ npx eslint src/*.ts
$`

Note that I added avoidEscape: true as well so it would allow e.g. 'foo "bar" baz' instead of requiring "foo \"bar\" baz".